### DISCUSSION TITLE
Post scan action event record

### DISCUSSION THREAD

**Discussion by mfaisalpasha** [trendmicro/cloudone-filestorage-plugins#64] [Q&A]
Hi everyone!
I am new to Cloud One. I want to know why in "cloudone-filestorage-plugins/post-scan-actions/aws-python-promote-or-quarantine/handler.py" there is a for loop for event['Records'] (Line number 114). In which scenario there will be more than one record in the SNS event?
Thank you in advance.

**Comment by carpusherw** [comment]
Hi @mfaisalpasha ,
You are correct according to this [doc](https://aws.amazon.com/sns/faqs/):

```
Q: Will a notification contain more than one message?
No, all notification messages will contain a single published message.
```

I suppose the author didn't aware of this in the first place and since it's an array, handling it with a loop made perfect sense.

**Reply by mfaisalpasha** [reply]
Thank you very much for your reply. Handling it with loop is perfect I think. As we want to skip rest of the actions when the scanner_status is not 0.
